### ISSUE TITLE
Last Line Variable

### ISSUE THREAD

**Issue by Ithrios** [joemcbride/outlander-osx#95]
Is there a variable that stores the last line matched by a waitfor/waitforre/match/matchre command? What I want to do is this:

put count my unsent
waitforre count out (1?[0-9]?[0-9]) uses remaining.

var number $matchedline
replacere("%number", "You count out (1?[0-9]?[0-9]) uses remaining.", "(1?[0-9]?[0-9])")

so that I can end up with a number that can be manipulated.

**Comment by joemcbride** [comment]
Should be able to do this with a matchre and matchwait.   Then in the matching label use $0, $1, etc. for the regex match groups.

**Comment by Ithrios** [comment]
Hmm. Well, that solved part of the problem. I was trying to use %0

For some reason, my script seems to be skipping the match wait function though.

My current script:
matchre end "You count out (1?[0-9]?[0-9]) use(s)? remaining."
put count my ungent
matchwait 10000000

end:
echo 0: %0

The results:

[Starting 'test' at 09:51 AM]
[test]: count my ungent
0:
[Script 'test' completed after 0.75 seconds total run time]
You count out 4 uses remaining.

As you can see it is returning 0: %0 before it gets the results from counting. Furthermore $0 empty, meaning that it isn't just a problem with the  UI. I have tried this with no number after the match wait and with matchwaitre and waitfor with exactly the same results.

**Comment by Ithrios** [comment]
Opps, had to correct my last comment.

I was using %0, so changing to $0 is helpful. Having said that for some reason my script refuses to respect waitfor commands. It seems to be ok with matchwait however. Maybe the feature only works with matchwait?

My current script:
put count my ungent
waitfor uses

end:
echo 0: $0

returns:
[Starting 'test' at 09:54 AM]
[test]: count my ungent
0: $0
[Script 'test' completed after 0.25 seconds total run time]
You count out 4 uses remaining.

alternate script:
matchre end "You count out (1?[0-9]?[0-9]) use(s)? remaining."
put count my ungent
matchwait 10000000

end:
echo 0: $0

**Comment by joemcbride** [comment]
Yes, currently that feature only works with `match/matchre/matchwait`.

You can use `matchre` with an if statement, like I do in the automapper script.  You can also pass arguments to the `gosub` command and access them with `$0-$n`, `n` being the number of arguments you passed.

```
top:
  if_1 then {
    gosub go %1

    if $powerwalk == 1 {
      gosub powerwalk
    }

    shift
    goto top
  }
  goto end

go:
  var dir $0
  var type default
  if matchre("%dir", "^(search|swim|climb|web|muck|rt|wait|slow|drag|script|room|ice) ") then
  {
    var type $1
    echo %type
    eval dir replacere("%dir", "^(search|swim|web|muck|rt|wait|slow|script|room|ice) ", "")
  }

  if %type = search {
    gosub %type
  }
```
